### ISSUE TITLE
Can't use URI type in the headers

### ISSUE THREAD

**Issue by kseo** [haskell-servant/servant#678]
The Servant documentation recommends using `Servant.Utils.Links.safeLink` for the Location header field.

```
-- ** 201 Created
--
-- Indicates that a new resource has been created. The URI corresponding to the
-- resource should be given in the @Location@ header field.
--
-- If the resource cannot be created immediately, use 'PostAccepted'.
--
-- Consider using 'Servant.Utils.Links.safeLink' for the @Location@ header
-- field.

-- | 'POST' with 201 status code.
--
type PostCreated = Verb 'POST 201
```

But it seems I can't use `URI` type for the header field because Servant requires `ToByteString` instance for the type. I can workaround by creating an orphan instance, but it is not ideal. Is there any recommended way to achieve this?

**Comment by phadej** [comment]
What version of `servant` are you using. `ToByteString` sounds quite old.

Though there aren't `ToHttpApiData` instance for `URI`, is there, @fizruk ?

**Comment by kseo** [comment]
I use servant-0.8.1/

**Comment by phadej** [comment]
Than I have to ask you to upgrade to `servant-0.9`, the `0.8` isn't supported.

If however you cannot upgrade, then you could define orphan `ToByteString URI` instance (or `newtype` wrapper with an instance).

There's e.g. `uriToString` in `Network.URI`.

**Comment by kseo** [comment]
Thanks!

I will upgrade the servant. It seems I can use `Link` which has `ToHttpApiData` instance.

**Comment by phadej** [comment]
for the record, there is `ToHttpApiData Link`, so there shouldn't be problem with `servant-0.10`.

@kseo, in `servant-0.9` we still return `URI` from `safeLink`. It will be fixed in 0.10 (to be released in a week or two).
